gui/odmr/odmrgui.py

```diff
@@ -585,10 +585,11 @@ def remove_ranges_gui_elements_clicked(self):
 
         # in case the removed range is the one selected for fitting right now adjust the value
         self._odmr_logic.ranges -= 1
-        max_val = self._odmr_logic.ranges
-        if self._odmr_logic.range_to_fit == max_val:
+        max_val = self._odmr_logic.ranges - 1
+        self._mw.fit_range_SpinBox.setMaximum(max_val)
+        if self._odmr_logic.range_to_fit > max_val:
             self._odmr_logic.range_to_fit = max_val
-            self._mw.fit_range_SpinBox.setMaximum(max_val)
+
```

Comment:
You have to set the new maximum for the SpinBox after this if-phrase. Otherwise this is causing some strange behavior for the odmrlogic's `range_to_fit`.